2-Operating-Systems/1-arch.md

```diff
@@ -0,0 +1,275 @@
+
+     ```
+     mkfs.vfat -F32 /dev/mmcblk0p1
+     ```
+
+  4. Encrypt and format the root partition
+
+     ```
+     # encrypt root partition with passphrase
+     cryptsetup -y -v luksFormat --type luks2 /dev/mmcblk0p2
+
+     # decrypt drive and expose as /dev/mapper/cryptroot
+     cryptsetup open /dev/mmcblk0p2 cryptroot
+
+     # make journaled ext4 partition in decrypted root device
+     mkfs.ext4 -j /dev/mapper/cryptroot
+     ```
+
+8. Mount the file systems
+
+  ```
+  mount /dev/mapper/cryptroot /mnt
+  mkdir /mnt/boot
+  mount /dev/mmcblk0p1 /mnt/boot
+  ```
+
+9. Select the Mirrors
+
+  ```
+  pacman -Sy pacman-contrib
+  curl https://www.archlinux.org/mirrorlist/all/https/ \
+  | sed 's/^#Server/Server/g' \
+  | rankmirrors -n 0 - \
+  > /etc/pacman.d/mirrorlist
+  ```
+  Note: This avoids manual mirror sorting. Native script does not yet exist.
+
+10. Install the base packages
+
+  ```
+  pacstrap /mnt base
+  ```
+
+11. Configure the System
+  1. Fstab: Generate 'File System TABle' of contents
+
+  ```
+  genfstab -U /mnt >> /mnt/etc/fstab
+  ```
+
+  2. Chroot: login to new arch installation by changing root
+
+  ```
+  arch-chroot /mnt
+  ```
+
+  3. Set time zone
+
+  ```
+  timedatectl set-timezone Region/City
+  hwclock --systohc
+  ```
+
+  4. Localization: setting your preferred language
+
+  ```
+  # Uncomment preferred language in /etc/locale.gen
+  sed -i 's/^#en_US/en_US/g' /etc/locale.gen
+
+  # Create /etc/locale.conf and set default language
+  echo "LANG=en_US.UTF-8" > /etc/locale.conf
+
+  # Non-US keyboard users can set their default layout
+  echo "KEYMAP=de-latin1" > /etc/vconsole.conf
+  ```
+
+  5. Name your computer!
+
+  ```
+  # Create the hostname file
+  hostnamectl set-hostname computername
+
+  # Specify new hostname for the network
+  echo "127.0.0.1 computername.localdomain computername localhost" > /etc/hosts
+  echo "::1 computername.localdomain computername localhost" >> /etc/hosts
+  ```
+
+  6. Initramfs: install system initialization bundle
+
+  ```
+  # Enable encryption support in initramfs
+  sed -i 's/ filesystems/ encrypt filesystems/g' /etc/mkinitcpio.conf
+
+  # Generate new initramfs
+  mkinitcpio -p linux
+  ```
+
+  7. Create a user with super user rights
+
+  ```
+  pacman -S sudo
+  echo "%wheel ALL=(ALL) ALL" >> /etc/sudoers
+  useradd -m -G wheel -s /bin/bash janedoe
+  passwd janedoe
+  ```
+
+  8. Configure and install bootloader
+
+  ```
+  bootctl --path=/boot install
+  echo "default arch" > /boot/loader/loader.conf
+  echo "title Arch Linux" >> /boot/loader/entries/arch.conf
+  echo "linux /vmlinuz-linux" >> /boot/loader/entries/arch.conf
+  echo "initrd /initramfs-linux.img" >> /boot/loader/entries/arch.conf
+  echo "options cryptdevice=UUID=$(blkid -o value /dev/mmcblk0p2 | head -n1):cryptroot root=/dev/mapper/cryptroot rw" >> /boot/loader/entries/arch.conf
+  ```
+
+  9. Install critical packages
+
+  ```
+  pacman -S dialog wpa_supplicant iw
```

Comment:
Again, these can be done as part of the pacstrap step.

Though note: these packages are only critical for wifi. For those of us with only ethernet connections, they're not even wanted!